src/poly/kzg/params.rs

```diff
@@ -298,81 +294,49 @@ where
         self.g_lagrange = g_to_lagrange(self.g.iter().map(|g| g.to_curve()).collect(), k);
     }
 
-    fn empty_msm(&'params self) -> MSMKZG<E> {
+    /// TODO
+    pub fn empty_msm(&'params self) -> MSMKZG<E> {
         MSMKZG::new()
     }
 
-    fn commit_lagrange(&self, poly: &Polynomial<E::Fr, LagrangeCoeff>, _: Blind<E::Fr>) -> E::G1 {
-        let mut scalars = Vec::with_capacity(poly.len());
-        scalars.extend(poly.iter());
-        let bases = &self.g_lagrange;
-        let size = scalars.len();
-        assert!(bases.len() >= size);
-        best_multiexp(&scalars, &bases[0..size])
-    }
-
     /// Writes params to a buffer.
-    fn write<W: io::Write>(&self, writer: &mut W) -> io::Result<()> {
+    pub fn write<W: io::Write>(&self, writer: &mut W) -> io::Result<()> {
         self.write_custom(writer, SerdeFormat::RawBytes)
     }
 
     /// Reads params from a buffer.
-    fn read<R: io::Read>(reader: &mut R) -> io::Result<Self> {
+    pub fn read<R: io::Read>(reader: &mut R) -> io::Result<Self> {
         Self::read_custom(reader, SerdeFormat::RawBytes)
     }
 }
 
-impl<'params, E: Engine + Debug> ParamsVerifier<'params, E::G1Affine> for ParamsKZG<E>
-where
-    E::G1Affine: SerdeCurveAffine<ScalarExt = <E as Engine>::Fr, CurveExt = <E as Engine>::G1>,
-    E::G1: CurveExt<AffineExt = E::G1Affine>,
-    E::G2Affine: SerdeCurveAffine,
-{
-}
-
-impl<'params, E: Engine + Debug> ParamsProver<'params, E::G1Affine> for ParamsKZG<E>
-where
-    E::G1Affine: SerdeCurveAffine<ScalarExt = <E as Engine>::Fr, CurveExt = <E as Engine>::G1>,
-    E::G1: CurveExt<AffineExt = E::G1Affine>,
-    E::G2Affine: SerdeCurveAffine,
-{
-    type ParamsVerifier = ParamsVerifierKZG<E>;
-
-    fn verifier_params(&'params self) -> &'params Self::ParamsVerifier {
+impl<'params, E: Engine + Debug> ParamsKZG<E> {
+    /// TODO
+    pub fn verifier_params(&'params self) -> &'params ParamsVerifierKZG<E> {
         self
     }
 
-    fn new(k: u32) -> Self {
+    /// UNSAFE function - do not use in production
+    pub fn new(k: u32) -> Self {
```

Comment:
We still need to be able to generate the parameters of the commitment scheme, otherwise we cannot do generic tests. I've moved that change to the other PR (here)[https://github.com/input-output-hk/halo2/pull/21/files#diff-ee3c2408bfbf948ebc0612573d2488bdbdcb044405261f1594fde49c26f26485R22-R26]. I'm not convinced of that, but I cannot think about a better solution.